curp/src/server/raw_curp/mod.rs

```diff
@@ -280,6 +283,48 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
         )
     }
 
+    /// Handle `shutdown` request
+    pub(super) fn handle_shutdown(
+        &self,
+        propose_id: ProposeId,
+    ) -> ((Option<ServerId>, u64), Result<(), ProposeError>) {
+        let st_r = self.st.read();
```

Comment:
Please use `map_read` and fetch the `leader_id`, `term` and `role` out. We can reduce the size of the critical section.

Reply:
We need to lock `st` and `log` at the same time, Otherwise, the term used by `log.push_shutdown` may be dirty data.